internal/locate/region_request.go

```diff
@@ -1200,7 +1200,11 @@ func (s *replicaSelector) updateLeader(leader *metapb.Peer) {
 				replica.attempts = maxReplicaAttempt - 1
 				replica.attemptedTime = 0
 			}
-			s.state = &accessKnownLeader{leaderIdx: AccessIndex(i)}
+			if !replica.deadlineErrUsingConfTimeout {
+				// if the new leader doesn't meet deadline exceeded error, switch to the new leader.
+				// Otherwise, keep the state unchanged and retry the request.
```

Comment:
This strategy may also need to be discussed with the `kv timeout feature` later. When timeout error is returned, there could be different situations, for example:
- The leader is busy or slow, it could not serve the request in time. In this case, trying follower using replica read is likely to fail too.
- There is some problem bettwen the kv-client and the leader peer. In this case, trying follower using replica read could be helpful.

To make better decisions maybe we could reference to the feedback slow information from the tikv nodes.

Reply:
Now that the above logic has been met, you can take a look at the test case.